static/js/ProfileDetail.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { FC } from "react";
+import { useParams } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+import { fetchProfile } from "./api/profiles";
+import DeleteProfileBtn from "./buttons/profiles/DeleteProfileBtn";
+import EditProfileBtn from "./buttons/profiles/EditProfileBtn";
+import OpenProfileListBtn from "./buttons/profiles/OpenProfileListBtn";
+import BaseLayout from "./components/BaseLayout";
+import NotificationRow from "./components/NotificationRow";
+import { queryKeys } from "./util/queryKeys";
+import useNotification from "./util/useNotification";
+import {
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+  Col,
+  Row,
+} from "@canonical/react-components";
+import { LxdDiskDevice, LxdNicDevice } from "./types/device";
+import { isDiskDevice, isNicDevice } from "./util/type";
+
+const ProfileDetail: FC = () => {
+  const { name } = useParams<{ name: string }>();
+
+  if (!name) {
+    return <>Missing name</>;
+  }
+
+  const notify = useNotification();
+
+  const {
+    data: profile,
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.profiles, name],
+    queryFn: () => fetchProfile(name),
+  });
+
+  if (error) {
+    notify.failure("Could not load profile details.", error);
+  }
+
+  if (isLoading) {
+    return <>Loading...</>; // TODO: improve, maybe with a nice loader?
+  } else if (!profile) {
+    return <>Could not load profile details.</>;
+  }
+
+  const btnProps = {
+    profile: profile,
+    notify: notify,
+    appearance: "",
+    className: "u-no-margin--bottom",
+    isDense: false,
+  };
+
+  const getUsedByNames = () => {
+    return profile.used_by?.map((path) => path.split("/").slice(-1)[0]);
+  };
+
+  const getCloudInitConfig = (
+    type: "user-data" | "vendor-data" | "network-config"
+  ) => {
+    return profile.config[`cloud-init.${type}`].replace("|\n", "");
+  };
+
+  return (
+    <BaseLayout
+      title={`Profile details for ${name}`}
+      controls={
+        <>
+          <EditProfileBtn label="Edit" {...btnProps} />
+          {profile.name !== "default" && (
+            <DeleteProfileBtn label="Delete" name={name} {...btnProps} />
+          )}
+          <OpenProfileListBtn />
+        </>
+      }
+    >
+      <NotificationRow notify={notify} />
+      <Row>
+        <table>
+          <thead>
+            <tr>
+              <th>Label</th>
+              <th>Value</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <th>Name</th>
+              <td>{profile.name}</td>
+            </tr>
+            <tr>
+              <th>Description</th>
+              <td>{profile.description}</td>
+            </tr>
+            <tr>
+              <th>Instances using this profile</th>
```

Comment:
I also recalled this information from our discussions, but then I could not find any evidence of this in the docs. It is my understanding that profiles can be applied to instances only, so representing something different in the UI would contradict what the user can find in the docs.

Or can you link me any reference in the docs where it says that a profile can be applied to anything that is not an instance?

For the link: sure, just keep in mind that the same shouldr be done in the instance overview tab, with profiles. I can take care of that in a future PR.

Reply:
True, would be good to link both ways.
Regarding entities, it might be undocumented. I remember it was mentioned that it actually not a common use case to link profiles to images, but to support windows it is used for attaching the TPM device. In that case the UI would be wired, because the heading says "Instances" but the entity is not an instance.

Reply:
Ok, but I think the point that we need to understand here is: does the back-end ever return anything that is not an instance in the `used_by` field here? Only in that case it would make sense to adapt the UI accordingly. It might be the case that, even if you can somehow link a profile to an image, that would not be returned in this list here.

Reply:
You might be right, this seems like an instance only used by at https://github.com/lxc/lxd/blob/master/lxd/profiles.go#L197

Let's keep it as is and remember to double check on the next core sync.

Reply:
Added links to instance details page @ [3b362fd](https://github.com/canonical/lxd-ui/commit/3b362fdf2a1568565b63d47c47b20775395fa671)